nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -250,7 +250,7 @@ in
             Restart = "always";
             AmbientCapabilities = [ "CAP_NET_ADMIN" "CAP_NET_RAW" "CAP_NET_BIND_SERVICE" ];
             ReadWritePaths = [ "/proc/sys/net/ipv6" ]
-              ++ lib.optionals useResolvConf [ "/etc/resolv.conf" "/run/resolvconf" ];
+              ++ lib.optionals useResolvConf ([ "/run/resolvconf" ] ++ config.networking.resolvconf.subscriberFiles);
```

Comment:
Perhaps we should expose the full list of required read/write paths?  Either by renaming `subscriberFiles` or by adding a new option?

Reply:
Uhm, let's try to keep it simple, for now: there's not much that interacts with resolvconf and it seems the only module using this subscriber mechanism is dnsmasq.

Before worrying about other unprivileged clients invoking `resolvconf`, I should deal witht the fact it doesn't set the right group on the generated files.

Reply:
> Before worrying about other unprivileged clients invoking resolvconf, I should deal with the fact it doesn't set the right group on the generated files.

Are you talking about something other than what this PR is fixing?  The files all have `resolvconf` as the group for me.

Reply:
Yes, unrelated to dnsmasq: the ownership of files in /run/resolvconf seem to use the wrong group.

Reply:
Ah, okay.  Just in case it's useful, this is what I see:

```
-rw-r----- 1 dhcpcd dhcpcd     115 Oct 17 13:24  /run/resolvconf/interfaces/enp4s0.dhcp
-rw-rw-r-- 1 root   resolvconf  23 Oct 15 23:23  /run/resolvconf/interfaces/static
-rw-rw-r-- 1 root   resolvconf 151 Oct 17 08:32  /run/resolvconf/interfaces/tun0
-rw-rw-r-- 1 root   resolvconf   2 Oct 15 23:23 '/run/resolvconf/metrics/0000001 static'
-rw-r----- 1 dhcpcd dhcpcd       2 Oct 17 13:24 '/run/resolvconf/metrics/0001002 enp4s0.dhcp'
```

Reply:
Yes, exactly: those written by dhcpcd have dhcpcd's own uid/gid.